src/ResourceManagement/ServerManagement/ServerManagement.Tests/ServerManagementTestBase.cs

```diff
@@ -0,0 +1,271 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Security.Principal;
+using System.ServiceProcess;
+using System.Threading.Tasks;
+using Microsoft.Azure.Management.ServerManagement;
+using Microsoft.Azure.Test.HttpRecorder;
+using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
+using Xunit.Abstractions;
+
+
+namespace ServerManagement.Tests
+{
+    public class ServerManagementTestBase : TestBase
+    {
+        protected static string ResourceGroup = "sdktest";
+        private static bool _once;
+
+        private readonly ITestOutputHelper _output;
+
+        protected static bool Recording
+        {
+            get { return HttpMockServer.Mode == HttpRecorderMode.Record; }
+        }
+
+        protected static bool ReuseExistingGateway
+        {
+            get
+            {
+                return Recording &&
+                       "true".Equals(Environment.GetEnvironmentVariable("SMT_REUSE_EXISTING_GATEWAY"),
+                           StringComparison.OrdinalIgnoreCase);
+            }
+        }
+
+        protected static string NodeName
+        {
+            get { return Environment.GetEnvironmentVariable("SMT_NODE_NAME"); }
+        }
+
+        protected static string NodeUserName
+        {
+            get { return Environment.GetEnvironmentVariable("SMT_NODE_USERNAME"); }
+        }
+
+        protected static string NodePassword
+        {
+            get { return Environment.GetEnvironmentVariable("SMT_NODE_PASSWORD"); }
+        }
+
+
+        protected static bool IsAdmin
+        {
+            get
+            {
+                return Extensions.Safe(
+                        () =>
+                            new WindowsPrincipal(WindowsIdentity.GetCurrent()).IsInRole(WindowsBuiltInRole.Administrator));
+            }
+        }
+
+        public ServerManagementTestBase(ITestOutputHelper output)
+        {
+            // add environment variables so that we can just run from VS to record.
+            Extensions.Default("TEST_HTTPMOCK_OUTPUT",
```

Comment:
Not sure what this code is trying to do.  Gneerally, you want to store default values you recall on playback as variables in the test recording.